pkg/provider/aci.go

```diff
@@ -536,6 +554,137 @@ func (p *ACIProvider) CreatePod(ctx context.Context, pod *v1.Pod) error {
 	return p.azClientsAPIs.CreateContainerGroup(ctx, p.resourceGroup, pod.Namespace, pod.Name, cg)
 }
 
+// get list of distinct acr servernames from pod
+func (p *ACIProvider) getImageServerNames(pod *v1.Pod) []string {
+	serverNamesMap := map[string]int{}
+	acrRegexp := "[a-z0-9]+\\.azurecr\\.io"
+	for _, container := range pod.Spec.Containers {
+		re := regexp.MustCompile(`/`)
+		imageSplit := re.Split(container.Image, -1)
+		isMatch, _ := regexp.MatchString(acrRegexp, imageSplit[0])
+		if len(imageSplit) > 1 && isMatch {
+			serverNamesMap[imageSplit[0]] = 0
+		}
+	}
+
+	serverNames := []string{}
+	for k := range serverNamesMap {
+		serverNames = append(serverNames, k)
+	}
+	return serverNames
+}
+
+func (p *ACIProvider) getManagedIdentityImageRegistryCredentials(pod *v1.Pod, identity *armmsi.Identity, containerGroup *client2.ContainerGroupWrapper) (*[]azaci.ImageRegistryCredential){
+	serverNames := p.getImageServerNames(pod)
+	ips := make([]azaci.ImageRegistryCredential, 0, len(pod.Spec.ImagePullSecrets))
+	if identity != nil{
+		for _, server := range serverNames {
+			cred := azaci.ImageRegistryCredential{
+				Server:  &server,
+				Identity: identity.ID,
+			}
+			ips = append(ips, cred)
+		}
+	}
+	return &ips
+
+}
+
+func (p *ACIProvider) setContainerGroupIdentity(identity *armmsi.Identity, identityType azaci.ResourceIdentityType, containerGroup *client2.ContainerGroupWrapper) {
```

Comment:
It would be great if we create new file (aka `identity.go`) and move these new func to it. Just to organize the code.